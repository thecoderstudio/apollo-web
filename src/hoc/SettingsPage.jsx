import React from 'react';
import styled from 'styled-components';
import SettingsSideNavigation from '../pages/settings/SettingsSideNavigation';
import media from '../util/media';

const ContentWrapper = styled.div`
  flex: 1 1 auto;
  display: grid;
  position: relative;
  grid-template-columns: [navigation] 300px [content] 1fr;
`;

const StyledNavigation = styled(SettingsSideNavigation)`
  grid-column: navigation;
  position: absolute;
  top: 0;
  bottom: 0;
`;

const ComponentWrapper = styled.div`
  grid-column: content;
  padding: 16px;
`;

function WithSettingsNavigation(Component) {
  return function wrap(props) {
    const { location } = props;
    return (
      <ContentWrapper>
        <StyledNavigation location={location} />
        <ComponentWrapper>
          <Component />
        </ComponentWrapper>
      </ContentWrapper>
    );
  };
}

export default WithSettingsNavigation;
