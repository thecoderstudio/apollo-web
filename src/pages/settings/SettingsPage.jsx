import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Switch, Route } from 'react-router-dom';
import SideNavigation from '../../components/SideNavigation';
import UserSettings from './UserSettings';
import PathAwareLink from '../../components/links/PathAwareLink';
import NotFound from '../NotFound';

const propTypes = {
  location: PropTypes.object.isRequired
};

const ContentWrapper = styled.div`
  flex: 1 1 auto;
  display: grid;
  position: relative;
  grid-template-columns: [navigation] 225px [content] 1fr;
`;

const StyledNavigation = styled(SideNavigation)`
  grid-column: navigation;
  position: absolute;
  top: 0;
  bottom: 0;
`;

const ComponentWrapper = styled.div`
  grid-column: content;
  padding: 16px;
`;

export default function SettingsPage(props) {
  const { location, match } = props;
  return (
    <ContentWrapper>
      <StyledNavigation>
        <PathAwareLink
          location={location}
          to='/settings/user_settings'
          iconClassName='fas fa-user-cog'
        >
          User settings
        </PathAwareLink>
      </StyledNavigation>
      <ComponentWrapper>
        <Switch>
          <Route exact path={`${match.url}/user_settings`} component={UserSettings} />
          <Route component={NotFound} />
        </Switch>
      </ComponentWrapper>
    </ContentWrapper>
  );
}

SettingsPage.propTypes = propTypes;
