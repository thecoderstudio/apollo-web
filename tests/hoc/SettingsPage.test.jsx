import React from 'react';
import renderer from 'react-test-renderer';
import SettingsPage from '../../src/hoc/SettingsPage';

function getComponent() {
  return renderer.create(
    <SettingsPage />
  );
}

describe("settings page", () => {
  const props = { location: { pathname: 'location' } };

  it("render correctly", () => {
    const tree = getComponent(props).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
