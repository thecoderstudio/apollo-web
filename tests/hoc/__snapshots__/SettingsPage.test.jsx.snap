// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings page render correctly 1`] = `null`;
